hw_diag/views/diagnostics.py

```diff
@@ -111,3 +113,40 @@ def version_information():
     }
 
     return response
+
+
+@DIAGNOSTICS.route('/v1/add-gateway-txn', methods=['POST'])
+def add_gateway_txn():
+    """
+    Generates an add_gateway_txn if a destination name and wallets are defined and valid.
+    Diagnostics report will be in the format below if successful.
+
+    https://docs.helium.com/mine-hnt/full-hotspots/become-a-maker/hotspot-integration-testing/#generate-an-add-hotspot-transaction
+    {
+        errors: [],
+        DESTINATION_ADD_GATEWAY_TXN_KEY: {
+            "address": "11TL62V8NYvSTXmV5CZCjaucskvNR1Fdar1Pg4Hzmzk5tk2JBac",
+            "fee": 65000,
+            "owner": "14GWyFj9FjLHzoN3aX7Tq7PL6fEg4dfWPY8CrK8b9S5ZrcKDz6S",
+            "payer": "138LbePH4r7hWPuTnK6HXVJ8ATM2QU71iVHzLTup1UbnPDvbxmr",
+            "staking fee": 4000000,
+            "txn": "CrkBCiEBrlImpYLbJ0z0hw5b4g9isRyPrgbXs9X+RrJ4pJJc9MkS..."
+        }
+    }
+    """
+    file = request.files['file']
```

Comment:
@posterzh - This looks like HTTP `multipart/form-data` style form POST. Where you are expecting the file to be present inside the file key. I was thinking more on the lines of reading the full raw POST data as the file content. On Flask side this means using `request.get_data()` or `request.data` instead of using `request.files` or `request.form`. See https://tedboy.github.io/flask/generated/generated/flask.Request.get_data.html#flask.Request.get_data

Reply:
Ok, it seems that `request.get_data()` can get the content of the raw file correctly without breaking the encoding. @kashifpk  Could you give show me how the client code that calls this end point should look like?

Reply:
Sure:

```python
import requests
my_data = 'your full payload with signature and actual json'.decode('utf8')  # converting to bytes from str
requests.post(url, data=my_data)
```